plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -128,15 +96,20 @@ def set_docker_image
       container.image = docker_path
     end
 
+    # helpful env vars, also useful for log tagging
     def set_env
       env = (container.env || [])
 
       # static data
       metadata = release_doc_metadata
-      [:REVISION, :TAG, :PROJECT, :ROLE, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
+      [:REVISION, :TAG, :DEPLOY_ID, :DEPLOY_GROUP].each do |k|
```

Comment:
- rev/tag/etc will only be added to the pods and not the deployments/services/selectors
 - existing tags will not be overirtten
 - template is responsible for setting consistent role/project labels everywhere

sounds good ?